lib/serializer/json_serializer.ex

```diff
@@ -31,7 +31,7 @@ defmodule Cashu.Serializer.JSON do
   def deserialize(<<"cashu", _version::binary-size(1), token::binary>>) do
     # if(version != "A", do: Logger.info("Got cashu token version #{version}"))
     case Base.url_decode64(token, padding: false) do
-      {:ok, json_str} -> deserialize(json_str, %TokenV3{})
+      {:ok, json_str} -> {:ok, deserialize(json_str, %TokenV3{})}
```

Comment:
hmm, looking at this again, the problem is we're not checking the `to_struct` function actually succeeds in creating a struct from the fields -- that function should return :ok or :error and bubble it up.

Reply:
I see. So, reading through [struct/2](https://hexdocs.pm/elixir/Kernel.html#struct/2), it will always succeed as non-existing keys are discarded and duplicates are dealt with. I ended up returning `{:ok, struct(...)}` from `to_struct`. Let me know if that's enough or we should do a bit more.